File: internal/content/testdata/bench/javascript/password-strength-check.js
function is_strong_password(s) {
  if (s.length < 8) return false;
  const hasUpper = /[A-Z]/.test(s);
  const hasLower = /[a-z]/.test(s);
  const hasDigit = /[0-9]/.test(s);
  const hasSpecial = /[^a-zA-Z0-9]/.test(s);
  return hasUpper && hasLower && hasDigit && hasSpecial;
}

File: internal/content/testdata/bench/rust/password-strength-check.rs
fn is_strong_password(s: String) -> bool {
    if s.chars().count() < 8 {
        return false;
    }
    let has_upper = s.chars().any(|c| c.is_uppercase());
    let has_lower = s.chars().any(|c| c.is_lowercase());
    let has_digit = s.chars().any(|c| c.is_ascii_digit());
    let has_special = s.chars().any(|c| !c.is_alphanumeric());
    has_upper && has_lower && has_digit && has_special
}
